sys/include/registry.h

```diff
@@ -0,0 +1,403 @@
+ */
+typedef struct _registry_instance_t registry_instance_t;
+
+/**
+ * @brief  Data structure of a configuration group see @p _registry_group_t.
+ */
+typedef struct _registry_group_t registry_group_t;
+
+/**
+ * @brief Data structure of a configuration parameter see @p _registry_parameter_t.
+ */
+typedef struct _registry_parameter_t registry_parameter_t;
+
+/**
+ * @brief Data types of the registry.
+ */
+typedef enum {
+    REGISTRY_TYPE_NONE = 0, /**< No type specified */
+
+    REGISTRY_TYPE_OPAQUE, /**< OPAQUE. */
+    REGISTRY_TYPE_STRING, /**< String. */
+    REGISTRY_TYPE_BOOL,   /**< Boolean. */
+
+    REGISTRY_TYPE_UINT8,  /**< 8-bits unsigned integer. */
+    REGISTRY_TYPE_UINT16, /**< 16-bits unsigned integer. */
+    REGISTRY_TYPE_UINT32, /**< 32-bits unsigned integer. */
+    REGISTRY_TYPE_UINT64, /**< 64-bits unsigned integer. */
+
+    REGISTRY_TYPE_INT8,  /**< 8-bits signed integer. */
+    REGISTRY_TYPE_INT16, /**< 16-bits signed integer. */
+    REGISTRY_TYPE_INT32, /**< 32-bits signed integer. */
+    REGISTRY_TYPE_INT64, /**< 64-bits signed integer. */
+
+    REGISTRY_TYPE_FLOAT32, /**< 32-bits float. */
+    REGISTRY_TYPE_FLOAT64, /**< 64-bits float. */
+} registry_type_t;
+
+/**
+ * @brief The type of a registry node.
+ * A registry node points to a namespace, schema, instance, group or parameter.
+ */
+typedef enum {
+    REGISTRY_NODE_NAMESPACE = 0,
+    REGISTRY_NODE_SCHEMA,
+    REGISTRY_NODE_INSTANCE,
+    REGISTRY_NODE_GROUP,
+    REGISTRY_NODE_PARAMETER,
+} registry_node_type_t;
+
+/**
+ * @brief A registry node represents a specific location within the registry configuration tree.
+ * It can point to a namespace, schema, instance, group or parameter.
+ * The configuration group and the configuration parameter contain a pointer to a
+ * schema instance, because they are children of a specific schema instance
+ * in the configuration tree.
+ */
+typedef struct {
+    registry_node_type_t type; /**< The type of the node */
+    union {
+        const registry_namespace_t *namespace; /**< Pointer to the configuration namespace */
+        const registry_schema_t *schema;       /**< Pointer to a configuration schema */
+        const registry_instance_t *instance;   /**< Pointer to a schema instance */
+        struct {
+            const registry_instance_t *instance; /**< Pointer to a schema instance */
+            const registry_group_t *group;       /**< Pointer to a configuration group */
+        } group;                                 /**< A configuration group depends on an instance */
+        struct {
+            const registry_instance_t *instance;   /**< Pointer to a schema instance */
+            const registry_parameter_t *parameter; /**< Pointer to a configuration parameter */
+        } parameter;                               /**< A configuration parameter depends on an instance */
+    } value;                                       /**< The location inside of the configuration tree */
```

Comment:
Generally I'm fine with these type of "long lines" where it's long because it tries to conform to a comment formatting but I'm not sure what the general consensus on this is (Maybe @crasbe for a short confirmation whether or not this counts as a long line)